vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,286 @@
+Push was rejected by remote source because the user does not have sufficient write permissions. 
+* Confirm that your user or user group has write permissions. In NuGet, the user must be at least a [Contributor role](https://learn.microsoft.com/azure/devops/artifacts/feeds/feed-permissions#permissions-table) to the feed.
+
+### Cause 2: Misconfigured NuGet Feed url
+<!--
+Steps to reproduce:
+1. Push a binary package to an existing NuGet feed with an incorrect URI.
+2. Set $env:VCPKG_BINARY_SOURCES="clear;nuget,<incorrect url>,readwrite"
+-->
+You might see the error:
+```
+System.Net.Http.HttpRequestException: Response status code does not indicate success: 405 (Method Not Allowed).
+```
+
+The server rejected NuGet's push request because it did not recognize the request method.
+* Verify the URI in your binary source. Ensure the URI directs to the service index of the feed, typically `<feed>/nuget/v3/index.json`.
+
+### Additional NuGet Resources
+Consult the [NuGet documentation](https://learn.microsoft.com/azure/devops/artifacts/nuget/publish) for guidelines on connecting to and publishing on a NuGet feed.
+
+## <a name="cache-upload"></a> Cache upload errors
+
+[!INCLUDE [debug](../../includes/debug.md)]
+
+You encounter errors uploading a binary package to your cache.
+
+### Cause 1: Binary cache provider failed to upload
+
+Uploads can fail for a variety of reasons, and error messages are usually provider specific.
+* Ensure you are authenticated to your cache.  Different providers authenticate differently.
+* Check if you've specified the correct URI for your cache.
+* Refer to [push troubleshooting](#push-failure) if you are using NuGet as a binary source.
+* Review documentation or the troubleshooting guide of your specific provider.
+
+## <a name="empty-cache"></a> Empty binary cache
+
+[!INCLUDE [debug](../../includes/debug.md)]
+
+Although you encountered no errors and vcpkg installation was successful, the binary cache remains empty. If you observed errors, consult [push troubleshooting for NuGet](#push-failure) and [upload troubleshooting for other providers](#cache-upload).
+
+### Cause 1: Vcpkg lacks write permissions to the binary cache
+<!--
+Steps to reproduce:
+1. Create a NuGet feed
+2. $env:VCPKG_BINARY_SOURCES="clear;nuget,C:/nugetfeed,read"
+3. Run ./vcpkg install rapidjson
+-->
+You are missing the following message in your output.
+```
+Uploading binaries for 'rapidjson:x64-windows' to <binary source> source <url>.
+Stored binaries in 1 destinations in 1.5 s.
+```
+
+Vcpkg skipped uploading the binary package to your binary cache.
+* Ensure your [binary cache configuration](binarycaching.md#configuration-syntax) is set to `write` or `readwrite`
+
+## Libraries rebuild instead of using remote binary cache
+
+[!INCLUDE [debug](../../includes/debug.md)]
+
+Your system rebuilds libraries locally, even when the required binary package is available in the remote binary cache.
+
+There is the absence of the following output:
```

Comment:
```suggestion
The following message is missing in your output.

```
We should keep type of phrases consistent. By using consistent phrasing, readers can use them as markers to skip to the next instance of the phrase. A different suggestion would be to do something like:

> **Diagnose**: The following message is missing in your output.